packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -407,6 +423,41 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
     return this._userPoolClientName;
   }
 
+  public get userPoolClientSecret(): SecretValue | undefined {
```

Comment:
Maybe I'm missing something here, but I don't think this can be undefined. You're either generating the `_userPoolClientSecret`, returning it as is if it already exists, or throwing an error.

Reply:
Agreed. Fixed